src/overlays/actors/ovl_Obj_Mure3/z_obj_mure3.c

```diff
@@ -31,41 +33,168 @@ const ActorInit Obj_Mure3_InitVars = {
     (ActorFunc)NULL,
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_8098F748[] = {
+static s16 sRupeeCounts[] = { 5, 5, 7, 0 };
+
+static InitChainEntry sInitChain[] = {
     ICHAIN_F32(uncullZoneForward, 100, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneScale, 1800, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneDownward, 100, ICHAIN_STOP),
 };
 
-#endif
-
-extern InitChainEntry D_8098F748[];
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure3/func_8098F040.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure3/func_8098F110.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure3/func_8098F220.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure3/func_8098F364.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure3/func_8098F438.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure3/ObjMure3_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure3/ObjMure3_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure3/func_8098F598.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure3/func_8098F5AC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure3/func_8098F5D0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure3/func_8098F5E4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure3/func_8098F66C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure3/func_8098F680.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure3/ObjMure3_Update.s")
+void func_8098F040(ObjMure3* this, PlayState* play) {
+    s32 i;
+    Vec3f spawnPos;
+
+    Math_Vec3f_Copy(&spawnPos, &this->actor.world.pos);
+    for (i = 0; i < ARRAY_COUNT(this->unk148) - 1; i++, spawnPos.y += 20.0f) {
+        if (!((this->unk164 >> i) & 1)) {
+            this->unk148[i] = (EnItem00*)Item_DropCollectible2(play, &spawnPos, 0x10001);
+            if (this->unk148[i] != NULL) {
+                this->unk148[i]->actor.room = this->actor.room;
+            }
+        }
+    }
+}
+
+void func_8098F110(ObjMure3* this, PlayState* play) {
+    s32 i;
+    Vec3f spawnPos;
+    f32 sin = Math_SinS(this->actor.world.rot.y);
+    f32 cos = Math_CosS(this->actor.world.rot.y);
+    f32 radius;
+
+    spawnPos.y = this->actor.world.pos.y;
+
+    for (i = 0, radius = -40.0f; i < ARRAY_COUNT(this->unk148) - 1; i++, radius += 20.0f) {
```

Comment:
I question whether ARRAY_COUNT is appropriate to use in this actor, given both that the logic here is written to assume that the number of elements is always 5, and that the array itself is reused to store an array of 6 elements